gradio/processing_utils.py

```diff
@@ -518,3 +522,61 @@ def _scale(a, n, m, copy=True):
 
 def strip_invalid_filename_characters(filename: str) -> str:
     return "".join([char for char in filename if char.isalnum() or char in "._- "])
+
+
+def ffmpeg_installed() -> bool:
+    return shutil.which("ffmpeg") is not None
+
+
+def video_is_playable(video_filepath: str) -> bool:
+    """Determines if a video is playable in the browser.
+
+    A video is playable if it has a playable container and codec.
+        .mp4 -> h264
+        .webm -> vp9
+        .ogg -> theora
+    """
+    try:
+        container = pathlib.Path(video_filepath).suffix.lower()
+        output = subprocess.run(
```

Comment:
Instead of using `subprocess`, could we use `ffmpeg.probe` for cleaner code and better error handling: https://kkroening.github.io/ffmpeg-python/#ffmpeg.probe